core/positions/engine.go

```diff
@@ -65,14 +70,54 @@ func New(log *logging.Logger, config Config, marketID string, broker Broker) *En
 	log = log.Named(namedLogger)
 	log.SetLevel(config.Level.Get())
 
-	return &Engine{
-		marketID:     marketID,
-		Config:       config,
-		log:          log,
-		positions:    map[string]*MarketPosition{},
-		positionsCpy: []events.MarketPosition{},
-		broker:       broker,
+	e := &Engine{
+		marketID:         marketID,
+		Config:           config,
+		log:              log,
+		positions:        map[string]*MarketPosition{},
+		positionsCpy:     []events.MarketPosition{},
+		broker:           broker,
+		updatedPositions: map[string]struct{}{},
```

Comment:
sorry, just saw you addressed it

Reply:
well, we usually use map / slice literals when not allocating, and make when needs allocation.

Reply:
@jeremyletang Composite literals will allocate just the same as when using `make` :) The difference between `make` and literals, other than the fact that in Go `make` is the canonical way, is that `make` let's you preallocate with specified size (`make(map[int]string, 10)`), while literals let you preallocate actual elements (`map[int]string{1: "1", 2: "2"}`). Otherwise, allocation is the same and performance is the same between `map[string]struct{}{}` and `make(map[string]struct{})`.

So the only way to not allocate in memory is by just declaring, like `var a []int` or `var m map[string]struct{}`. With maps, you should always instantiate them, or you get a `nil` pointer at some point.

Reply:
Mate I know that. No need for the lesson. And that’s the reason why I say we use literals here because we don’t needs to allocate a size that we know about, and the reason why I use literal. Per the guidelines which points to the fo guideline which also if not mistaken say the right ways, we always use literals when we do not know of how much we want to allocate, and make when we know (e.g copying a slice, a map, or when we know the size before)

Reply:
@jeremyletang I'm sorry, didn't mean to come off as a smartass. I misunderstood when you said you use literals when not allocating, for some reason I thought you were talking about the actual allocation of the slice/map in memory, and not the size/elements...
Now I see what you meant, it makes sense.

Reply:
No worries.